chain/chain/src/metrics.rs

```diff
@@ -155,3 +155,51 @@ pub(crate) static LARGEST_FINAL_HEIGHT: Lazy<IntGauge> = Lazy::new(|| {
     )
     .unwrap()
 });
+
+pub(crate) enum ReshardingStatus {
+    /// The StateSplitRequest was send to the SyncJobsActor.
+    Scheduled,
+    /// The SyncJobsActor is performing the resharding.
+    BuildingState,
+    /// The resharding is finished.
+    Finished,
+}
+
+impl Into<i64> for ReshardingStatus {
```

Comment:
nit: https://doc.rust-lang.org/std/convert/trait.From.html
> One should always prefer implementing From over [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) because implementing From automatically provides one with an implementation of [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) thanks to the blanket implementation in the standard library.